toolchain/check/convert.cpp

```diff
@@ -496,7 +498,7 @@ static auto ConvertStructToStructOrClass(Context& context,
         ConvertAggregateElement<SemIR::StructAccess, TargetAccessInstT>(
             context, value_loc_id, value_id, src_field.type_id, literal_elems,
             inner_kind, target.init_id, dest_field.type_id, target.init_block,
-            src_field_index);
+            src_field_index, src_field_index + dest_has_vptr);
```

Comment:
```suggestion
            src_field_index, src_field_index + dest_has_vptr ? 1 : 0);
```

Whenever I see code that treats a boolean as a number, I have to stop and think through what the rules for that are.

Reply:
Ah, sure - there was another use of this already, so I addressed both by introducing a `dest_vptr_offset` integer local to use in both places. b68ab0d6cb171bd30af67d143424169353ec327c